src/initial_conditions/initial_conditions.jl

```diff
@@ -192,7 +192,20 @@ struct MoistFromFile <: InitialCondition
     file_path::String
 end
 
-function (initial_condition::MoistFromFile)(params)
+"""
+    WeatherModel(start_date, start_time)
+
+An `InitialCondition` that initializes the model with an empty state, and then overwrites
+ it with the content of the weather model. We assume that the weather initial condition 
+ is stored in a correctly named and formatted file in some TODO artifact. 
+"""
+struct WeatherModel <: InitialCondition
+    start_date::String
+    start_time::String
+end
+
+
+function (initial_condition::Union{MoistFromFile, WeatherModel})(params)
     function local_state(local_geometry)
```

Comment:
Is there a reason why we can't do
```suggestion
function (initial_condition::InitialCondition)(params)
    function local_state(local_geometry)
```

Note that I am assuming that `InitialCondition` is an abstract type.